### PR TITLE
feat(cli): route session actions through gmuxd; add --host/@suffix; local-default --list

### LINKED ISSUE #221: CLI: `--host` / `@suffix` for all session subcommands; strict local-default for consistency
## Motivation

Today the CLI splits the world cleanly into "things that work on local sessions" and "things that error on peer sessions." `--send`, `--tail`, `--attach`, `--kill`, `--wait` all bail out with `only supported for local sessions` the moment a peer is involved. `--list` is the only exception — it federates by default.

This creates a frustrating discontinuity in agent-driven workflows. Concrete example from today: I had to drive a `chwd -a` install on a peer machine. With federation enabled I could see the session in `gmux --list`, but I couldn't `--send` answers into it. End result: had to re-enable SSH for the duration of the work. Federation was technically there but felt one-way.

The daemon-side mechanism already exists — `peer.Forward` and `peer.ProxyWS` are wired and the web UI uses them for attach/kill/resume/read/restart. The CLI just shortcuts to session sockets and never goes through that path.

## Proposal

Every gmux CLI subcommand that takes a session ID accepts both:

1. A fully-qualified `id@host` (canonical form), or
2. A `--host=<name>` flag with a short ID.

Subcommands that don't take a session ID (like `--list`, `gmux <cmd>` to create) accept `--host=<name>` to scope or target the operation.

**Consistency rule**: with no `--host` and no `@host` suffix, every subcommand operates **strictly locally**. No magic peer fallback. If you want to look across peers, opt in with `--all`.

### Examples

```bash
# Address a peer session (both forms equivalent)
gmux --send c0b3c1a1@konyvtar 'foo'
gmux --send --host=konyvtar c0b3c1a1 'foo'

gmux --tail 100 c0b3c1a1@konyvtar
gmux --kill c0b3c1a1@konyvtar
gmux --attach c0b3c1a1@konyvtar
gmux --wait c0b3c1a1@konyvtar

# Create a new session on a peer (closes today's biggest gap — the
# one that drove me back to SSH for the chwd install)
gmux --host=konyvtar fish
gmux --no-attach --host=konyvtar bash -c 'sudo chwd -a; sleep 30'

# Discovery
gmux --list                # local only (new default)
gmux --list --all          # local + all peers
gmux --list --host=konyvtar  # only konyvtar
```

### `--list` output format

The `@host` info moves into the ID column so the ID is a single copy-paste unit:

```
c0b3c1a1@konyvtar  alive  shell   ~ - fish
10da1bcc           alive  pi      ...
```

(Local sessions stay bare.)

### Error UX

When a short ID has no local match but a peer has it, the error includes a "did you mean" hint:

```
$ gmux --send c0b3c1a1 'foo'
gmux: session c0b3c1a1 not found locally. Did you mean c0b3c1a1@konyvtar?
```

This keeps the strict-consistency rule (`--send c0b3c1a1` is still an error) while rescuing the most common case (the user copy-pasted from `--list --all` and forgot the suffix).

## Out of scope

- **`gmuxd` subcommands** (`status`, `restart`, `auth`, `remote`, `log-path`) stay local-only. They manage *daemon* state, not sessions. If you want to manage a remote daemon, do it via `gmux --host=peer systemctl --user restart gmuxd`. Keeps the security model clean: peers manage their own daemons.
- **Action fan-out**. `--kill --all c0b3c1a1` is not supported. `--all` is only for discovery (`--list`). Mass actions are a footgun (one typo, a lot of dead sessions).

## Backward compatibility

One real break: `gmux --list` becomes local-only by default. Federated requires `--all`. Worth a clear changelog note; gmux is pre-1.0 and the user base is small enough that a hard break is cleaner than a deprecation period that leaves stderr noise for releases.

Everything else is purely additive — today's peer-session operations error, so making them succeed via `--host`/`@suffix` doesn't break anyone.

## Why this is the right shape

The two design tensions are addressing form (flag vs suffix) and default behavior (local vs federated):

- **Both forms** (`--host=` and `@suffix`) because the suffix is already the wire format gmuxd uses internally and what shows up in URLs and in `--list`, so it should just work in commands. `--host=` is the right shape for create-style subcommands that don't take an ID.
- **Strict local default** because session lookup is almost always local, and when you're looking remotely you usually know which host. Consistency wins over magic.

## Implementation notes

- For peer-targeted operations, the CLI should route through the local gmuxd's HTTP/WS proxy (`peer.Forward` + `peer.ProxyWS`) rather than shortcutting to the session's unix socket. Local sessions can keep the direct-socket fastpath.
- The CLI already short-circuits early in [`cli/gmux/cmd/gmux/actions.go`](https://github.com/gmuxapp/gmux/blob/main/cli/gmux/cmd/gmux/actions.go) when `sess.SocketPath == ""`. The fix is to fall through to a gmuxd-routed implementation instead of erroring.
- `--list` "did you mean" hint requires the CLI to ask gmuxd for peer matches when local lookup fails; gmuxd already has the namespaced session inventory.
- `--host=<name>` resolves against configured peers in `host.toml` plus auto-discovered tailnet peers. Unknown name is a hard error (no JIT discovery — that's a separate feature if ever needed).

## Why I'm filing this

Today's workflow used SSH as a fallback three times because gmux federation wasn't actionable from the CLI. With this change, gmux federation becomes a complete control plane and SSH becomes optional for agent-driven work. That's the pitch tsnet remote was making in the first place.

### LINKED ISSUE #222: `gmux --tail` should read from on-disk scrollback when the live socket is gone (dead sessions)
## Reproducer

```bash
id=$(gmux --no-attach bash -c 'echo "important error output"; echo "more context"; exit 1')
sleep 1

# The session shows up correctly as dead
gmux --list | grep ${id#sess-}
# f977d28b  dead   shell   bash -c echo "important error output"; echo "more context"; exit 1

# But --tail fails with a raw socket dial error
gmux --tail 20 $id
# gmux: Get "http://session/scrollback/tail?n=20": dial unix /tmp/gmux-sessions/sess-f977d28b.sock: connect: connection refused

# Meanwhile, the scrollback is sitting on disk untouched
cat ~/.local/state/gmux/sessions/$id/scrollback
# important error output
# more context
```

## Why this hurts

This is exactly the case where `--tail` is most valuable: **a session that died fast and you want to know why.** The agent-driven flow today reproduced it three times: I'd launch something with `gmux --no-attach`, it would fail before I could attach, and then I'd be unable to retrieve the error message through the CLI. I ended up `cat`ing `~/.local/state/gmux/sessions/<id>/scrollback` manually each time.

The data the user wants is *literally on disk, in plaintext, in a known location*. The CLI just doesn't know to look there when the live socket is gone.

## What's happening

In [`cmdTail` (cli/gmux/cmd/gmux/actions.go)](https://github.com/gmuxapp/gmux/blob/main/cli/gmux/cmd/gmux/actions.go):

```go
if sess.SocketPath == "" {
    switch {
    case sess.Peer != "":
        // ... "only supported for local"
    case !sess.Alive:
        fmt.Fprintf(os.Stderr, "gmux: session %s is not running\n", ...)
    default:
        // ... "no socket path"
    }
    return 1
}

client := sessionSocketClient(sess.SocketPath)
// ...dials sess.SocketPath...
```

For a dead session, `sess.Alive` is `false` but `sess.SocketPath` is still populated (gmuxd doesn't clear it on session death). So the `!sess.Alive` arm — which prints the friendly `"session is not running"` message — never fires. Instead the code falls through to a dial against a unix socket that no longer exists, and surfaces the raw `connect: connection refused` error to the user.

## Proposed fix

`gmux --tail` should read from the on-disk scrollback when the live socket isn't reachable. The data path is well-defined: `<gmuxd-state>/sessions/<sess-id>/scrollback` (the runner persists scrollback there on exit, and gmuxd already manages that directory).

Suggested behavior:

1. Live path first (current): dial `sess.SocketPath`, stream from `/scrollback/tail`.
2. Fallback: if dial fails (or `!sess.Alive`), read the on-disk scrollback file and emit the last N lines.
3. Only error with a hard "not found" message if both paths are unavailable (e.g. session record was dismissed and disk file is gone).

The on-disk file is raw PTY bytes including ANSI — exactly the same format the live `/scrollback/tail` endpoint already streams — so no transformation is needed beyond a "tail last N lines" pass.

Peer sessions are out of scope here (they'll be addressed separately by #221), but the same pattern would apply: try the peer's scrollback endpoint via gmuxd `ProxyWS`, fall back to a forwarded read of the peer's on-disk scrollback if the runner is dead. That's a follow-up.

## Why this is worth fixing despite the workaround

The state directory layout is an internal contract — telling users "just `cat ~/.local/state/gmux/sessions/<id>/scrollback`" couples them to that path forever. Putting the fallback in `--tail` keeps the contract internal, makes the CLI feel complete (it already promises "the last N lines of a session"), and removes the need to remember a magic filesystem location for the most common diagnostic case.

### PR DESCRIPTION
**Pull request by mgabor3141**
Closes #221, closes #222.

## What this changes

Routes every session-targeted CLI subcommand (`--tail`, `--send`, plus `--list` and the new `--host` / `@suffix` plumbing) through gmuxd's HTTP API instead of dialing the session's unix socket directly.

The two open issues are different symptoms of the same architectural smell: the CLI was shortcut-dialing per-session sockets, which works for live local sessions and falls apart for everything else (dead → socket gone; peer → can't reach it). Once the CLI talks only to gmuxd, the "where does this session live?" question lives in exactly one place, and both bugs collapse.

| Subcommand | Before | After |
|---|---|---|
| `--tail` | dial session socket; refuses dead/peer sessions | `GET /v1/sessions/<id>/scrollback?tail=N`; works for live, dead, peer-live, peer-dead |
| `--send` | dial session socket; refuses peer sessions | `POST /v1/sessions/<id>/input`; routes through `peer.Forward` for peer sessions |
| `--list` | federated by default | local by default; `--all` federates; `--host=<peer>` scopes (the one breaking change in this PR) |

Plus the user-facing addressing layer that this enables:

- **`--host=<peer>` flag** on every session action, and **`id@<peer>` suffix** as a canonical equivalent. Both forms reconcile cleanly (conflict error if both given and they disagree).
- **Strict local-default** in `matchSession`: with no host hint, peer sessions are invisible to lookup. A short ID that only matches a peer session produces `gmux: session "abc" not found locally. Did you mean abc@konyvtar?` rather than a misleading "not found".
- **`@host` moves into the ID column** of `--list` output so each row's ID is a single copy-paste unit that works directly with `--send` / `--kill` / `--tail`.
- **Error messages keep the `@peer` suffix** (via a new `displayID` helper). Otherwise an error like "session 307463e2 not found" would be ambiguous between a local session and the peer-owned one the user typed.

The runner's previously-undocumented `GET /scrollback/tail` endpoint (the one the CLI used to shortcut to) is removed; ADR 0005 establishes the runner socket as a private implementation detail of gmuxd. Its rendering helpers are lifted into `packages/scrollback` as `RenderTail` and used by gmuxd's broker, so `--tail` output is byte-identical to pre-PR (rendered plain text, ANSI stripped, cursor overwrites collapsed). **No user-visible format change for `--tail`.**

## Breaking change

Just one: `gmux --list` no longer shows peer sessions by default. Pass `--all` for the previous behavior, or `--host=<peer>` to scope to one peer. Marked with `feat(cli)!:` so git-cliff surfaces it in the release notes.

Scripts that depended on `gmux --list` returning federated output (e.g. `gmux --list | wc -l` counting all sessions across peers) get a smaller result after upgrade with no error — the kind of change that warrants the conventional-commits `!` even pre-1.0.

## Bug fix caught during review

`apiclient.ForwardAction` was dropping the original request's query string when forwarding to a peer. With the new `?tail=N` parameter on the scrollback broker, a cross-peer `gmux --tail 5 <id>@<peer>` would have silently downloaded the entire scrollback instead of the last 5 lines. Fixed in its own commit (`fix(daemon): preserve query strings when forwarding session actions to peers`) with a regression test that's mutation-verified to actually fail when the fix is reverted.

Found this by re-reading `ForwardAction` while sanity-checking that the new query-param surface I added would actually traverse peer boundaries. Worth flagging: there's no two-gmuxd integration test in this PR, so the test that catches this is the unit test in `apiclient`, not an end-to-end one. A real cross-peer `--tail` smoke would need both ends running this branch.

## Out of scope (deferred follow-ups, explicitly rejected at parse time)

- **Remote create** (`gmux --host=peer fish`): the gmuxd-side endpoint for creating sessions from a peer doesn't exist yet. CLI rejects with a clear "planned for a follow-up" message.
- **`--wait` cross-peer**: needs peer-side Status event streaming, which isn't wired. CLI rejects with "does not yet support --host".

## Architecture note

ADR 0005 (`docs/adr/0005-cli-routes-through-gmuxd.md`) captures the trade-off (one extra ~1ms HTTP hop on every local action, in exchange for the live/dead/peer routing collapsing into one place inside gmuxd), the alternatives considered, and the one duplication added on purpose (the `maxInputBytes` cap is now defined in both runner and gmuxd, so the 413 surfaces at gmuxd's edge instead of silent truncation in the runner).

## How `--tail` produces plain text

The runner used to render `--tail` output from its in-memory `vt.Emulator` cell grid. Removing the runner endpoint and having gmuxd render from disk needed a terminal emulator on the gmuxd side. `charmbracelet/x/vt` was already a runner dependency, so adding it to `packages/scrollback` is cheap. `RenderTail` replays raw PTY bytes through a fresh emulator at the session's last-known dimensions (falling back to 80x24) and returns the trailing N lines as plain text. The rendering covers ANSI (interpreted as styling, not text), cursor overwrites (collapsed to the final visible content), and idle-TUI blank-row trimming. Net effect: `gmux --tail` output is unchanged from pre-PR.

## How it was tested

- **Unit tests**: full coverage for `RenderTail` (ANSI stripping, cursor-overwrite collapse, line limiting, blank-row trimming, edge cases including non-positive dimensions and broken readers), the `?tail=N` parameter on the scrollback broker (end-to-end ANSI strip + cursor overwrite assertions, content-type, IO-error → 500), `discovery.SendInput` (delivers body verbatim, surfaces runner errors), `ForwardAction` query-string preservation, and all the new `matchSession` cases (strict-local default, friendly hint, `@suffix`, `--host`, conflict, agreement, empty-suffix rejection, multi-peer ambiguity hint). 27 packages pass, 0 failures.
- **Mutation-verified** the two highest-leverage new tests (`TestForwardAction_PreservesQueryString` and `TestMatchSessionFriendlyHintForPeerOnlyMatch`) by temporarily breaking the implementation and confirming the test actually fails.
- **End-to-end on real hardware**: a parallel gmuxd running against this branch, with the real session state dir symlinked in. Confirmed:
  - `--tail` on a dead session that printed `\x1b[31mRED LINE\x1b[0m\nplain line\nloading...\rDONE\n` returns exactly `RED LINE\nplain line\nDONE\n` — ANSI gone, `loading...` overwritten, `\r` collapsed.
  - The raw (no-tail) broker output is byte-identical to `cat <state>/sessions/<id>/scrollback`, so the web UI path is unaffected.
  - `--send` delivers single messages and piped multi-line stdin.
  - `--list` strict-local default, `--list --all` federated, `--list --host=konyvtar` scoped.
  - "Did you mean abc@konyvtar?" hint fires for peer-only short IDs.

A true cross-peer `--send` / `--tail` smoke needs both ends on this branch and isn't part of the test evidence here. The routing path is unit-tested at every join.

## Commits

```
refactor: render gmux --tail output through terminal emulator in gmuxd
fix(cli): list candidates on multi-peer match; reject empty @host suffix
fix(daemon): return 500 when TailBytes fails, not silent 200
refactor(cli): drop runner's GET /scrollback/tail endpoint
docs(adr): 0005 — CLI routes session actions through gmuxd
refactor(cli): show @peer suffix in error messages via displayID
fix(daemon): preserve query strings when forwarding session actions to peers
feat(cli)!: route session actions through gmuxd; add --host/@suffix; local-default --list
feat(daemon): /v1/sessions/<id>/input action enables cross-peer --send
feat(daemon): tail=N query param on scrollback broker for line-based tailing
feat(scrollback): add TailBytes helper for line-based scrollback tailing
```

### CODE REVIEW

File: services/gmuxd/cmd/gmuxd/scrollback.go [hunk 1]
```diff
@@ -69,6 +90,23 @@ func scrollbackBrokerHandler(
...
+		// and lets us produce the response in a single Write.
+		tail, err := scrollback.TailBytes(rc, tailN)
+		if err != nil {
+			log.Printf("scrollback tail: %s: %v", sessionID, err)
+			// Headers were already sent (200 + Content-Type). The best
+			// we can do is end the response; the client sees a short
+			// read and surfaces it. Logging here is what makes the
+			// failure debuggable.
+			return
+		}
```
[the remark this answers is not in the document]
Comment (mgabor3141) [reply services/gmuxd/cmd/gmuxd/scrollback.go]: Fixed in d8e1572 (`fix(daemon): return 500 when TailBytes fails, not silent 200`). The misleading comment is gone, the handler now `writeError`s a 500. Added `TestBrokerTailParamIOErrorReturns500` which exercises specifically the TailBytes error path (by seeding the scrollback location with a directory so OpenReader succeeds but ReadAll fails with EISDIR); mutation-verified that reverting the fix makes the test fail.

File: cli/gmux/cmd/gmux/actions.go [hunk 2]
```diff
@@ -58,63 +55,140 @@ func fetchSessions() ([]cliSession, error) {
...
+		for _, s := range sessions {
+			if s.Peer != "" {
+				peerPool = append(peerPool, s)
+			}
+		}
+		if hint, _ := lookupInPool(peerPool, ref); hint != nil {
+			return cliSession{}, fmt.Errorf("session %q not found locally. Did you mean %s@%s?",
+				ref, shortID(hint.ID), hint.Peer)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (mgabor3141) [reply cli/gmux/cmd/gmux/actions.go:152]: Fixed in 7fd874b (`fix(cli): list candidates on multi-peer match; reject empty @host suffix`). Multi-peer match now lists the qualified candidates so the user sees what to type. Took the spirit of your suggestion but emit the actual candidate list rather than a generic "qualify with id@<peer>" message, since picking one to highlight would silently favor an arbitrary peer. New test `TestMatchSessionMultiplePeerMatchesGetCandidateList` uses a realistic prefix-match scenario (full IDs are globally unique, but a typed prefix can match across peers).

File: cli/gmux/cmd/gmux/actions.go [hunk 3]
```diff
@@ -58,63 +55,140 @@ func fetchSessions() ([]cliSession, error) {
...
-	// resolves to a prefix when a session is literally named "abcd".
+	// Split off any @host suffix. Reconcile with the --host flag.
+	if idx := strings.LastIndex(ref, "@"); idx > 0 {
+		suffixHost := ref[idx+1:]
+		if host != "" && host != suffixHost {
+			return cliSession{}, fmt.Errorf("--host=%q conflicts with %q in session ref", host, suffixHost)
+		}
+		host = suffixHost
+		ref = ref[:idx]
+	}
```
[the remark this answers is not in the document]
Comment (mgabor3141) [reply cli/gmux/cmd/gmux/actions.go:107]: Fixed in 7fd874b. `id@` now errors with `session ref "id@" has empty @host suffix` rather than silently scoping to local. New `TestMatchSessionEmptyHostSuffixRejected` covers it.

**Comment by mgabor3141** [comment]
@greptile

File: cli/gmux/cmd/gmux/actions.go [hunk 4]
```diff
@@ -58,63 +55,158 @@ func fetchSessions() ([]cliSession, error) {
...
+			// as local would silently scope wrong; demand the user make
+			// the intent explicit.
+			return cliSession{}, fmt.Errorf("session ref %q has empty @host suffix", ref)
+		}
+		if host != "" && host != suffixHost {
+			return cliSession{}, fmt.Errorf("--host=%q conflicts with %q in session ref", host, suffixHost)
+		}
+		host = suffixHost
+		ref = ref[:idx]
+	}
```
[the remark this answers is not in the document]
Comment (mgabor3141) [reply cli/gmux/cmd/gmux/actions.go:107]: Not reachable in practice. Every code path that sets `cliSession.Slug` runs the value through one of the two `Slugify` helpers:

- `packages/adapter/adapter.go:71` (`Slugify`): regex `[^a-z0-9]+` replaces non-alphanumeric with `-`, so `@` becomes `-`.
- `services/gmuxd/internal/projects/projects.go:305` (`IsValidSlug`): regex `^[a-z0-9]+(-[a-z0-9]+)*$` would reject `@` outright.

The runner's `PUT /slug` endpoint accepts arbitrary bytes, but `rg -n 'PUT.*\"/slug\"|"/slug"'` shows it has zero callers anywhere in the codebase — the slug invariant ("URL-safe kebab-case alphanumeric") holds at every actual writer. The hypothetical `fix@home` slug isn't producible by any existing path.

I'll leave the split-on-`@` as-is rather than adding a defensive "split only when suffix matches a known peer" check: the AGENTS.md guidance is to encode invariants where they're established (in `Slugify`) rather than re-checking them at every consumer.

If a future change ever loosens slug validation to allow `@`, this will need to be revisited — but that's a change to the slug invariant, and a test like "every reachable Slug ends up Slugify-shaped" would catch it then.
